### PR TITLE
(Feature) Claim should be triggered automatically when user is first whitelisted

### PR DESCRIPTION
**Pull request by alt-a1r**
# Description

fixed styles for claim button and fv-modal

About #231

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/good-design/src/core/buttons/ActionButton.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 
 interface IBasicButtonProps extends IButtonProps {
   text: string;
   onPress: () => {};
 }
 
-const ButtonAction = ({ text, ...props }: IBasicButtonProps) => <Button {...props}>{text}</Button>;
+const ButtonAction = ({ text, ...props }: IBasicButtonProps) => {
+  const Button = Factory(NBButton, theme)
+  return <Button {...props}>{text}</Button>
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/buttons/ActionButton.tsx:12]:
```suggestion

  return <Button {...props}>{text}</Button>
```

Comment (johnsmith-gooddollar) [reply packages/good-design/src/core/buttons/ActionButton.tsx:12]: why this is need?

File: packages/good-design/src/core/buttons/ClaimButton.tsx [hunk 2]
```diff
@@ -44,25 +46,28 @@ function ClaimButton({ firstName, method, refresh, ...props }: FVFlowProps) {
...
+    openClaimModal();
+  }, [isVerified, claimAmount, openClaimModal]);
 
   return (
     <View {...props}>
-      <View flex={1} alignItems="center" justifyContent="center">
-        <FvModal method={method} isOpen={showModal} onClose={handleClose} firstName={firstName} />
-      </View>
       <ActionButton text={buttonTitle} onPress={handleClaim} />
+
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/buttons/ClaimButton.tsx:58]:
```suggestion
```

File: packages/good-design/src/core/modals/FVModal.tsx [hunk 3]
```diff
@@ -87,4 +90,13 @@ function FVModal({ firstName, method, onClose = noop, ...props }: FVModalProps)
   );
 }
 
-export default FVModal;
+export const theme = {
+  baseStyle: withThemingTools(({ colorModeValue }: { colorModeValue: any }) => ({
+    justifyContent: "center",
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/modals/FVModal.tsx:95]: only fonts / colors / background / border colors  / shadow props are allowed in theme
please adjust all styles

check this as reference https://github.com/GoodDollar/GoodWeb3-Mono/pull/27/commits/3756062203e316edcd6e05a8bad616900930fcb6

File: packages/good-design/src/core/modals/FirstClaimModal.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import CrossIcon from "./cross";
+import { openLink } from "../utils/linking";
+import ButtonAction from "../buttons/ActionButton";
+import { FVModalProps } from "../buttons/ClaimButton";
+import { withTheme, withThemingTools } from "../../theme";
+
+export interface FirstClaimModalProps extends IModalProps {}
+
+const FirstClaimModal: FC<FirstClaimModalProps> = ({ onClose = noop, ...props }) => {
+  return (
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/modals/FirstClaimModal.tsx:15]: if no additional logic (callbacks, hooks etc) - do not use return

e.g. const SimpleComponent = props => (
  jsx tree here
)

File: packages/good-design/src/hooks/useModalState.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { useCallback, useState } from "react";
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/hooks/useModalState.ts:1]: see modal concept draft here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/27#pullrequestreview-1182907541

important: should be discussed with @sirpy

File: packages/good-design/src/hooks/useQueryParam.ts [hunk 6]
```diff
@@ -1,4 +1,13 @@
...
-}
+  const { search, hash } = window.location;
+
+  return useMemo(() => {
+    // in case of hash we have empty search field
+    const queryString = search || hash?.split("?")[1];
+    const params = new URLSearchParams(queryString);
+
+    return params.get(param);
+  }, [search, hash, param]);
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/hooks/useQueryParam.ts:12]:
```suggestion
  }, [param]);
```

non-state vars / props / anything derived from them shouldn't be deps

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/hooks/useModalState.ts (hunk 5 above)
Comment (johnsmith-gooddollar) [reply packages/good-design/src/hooks/useModalState.ts:1]: @roma-hladilka 
I saw you merged @sirpy 's modal implementation
but you still use your own
please fix this and we will bet able to merge

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/core/buttons/ActionButton.tsx (hunk 1 above)
Comment (roma-hladilka) [reply packages/good-design/src/core/buttons/ActionButton.tsx]: changed

File: packages/good-design/src/hooks/useModalState.ts (hunk 5 above)
Comment (roma-hladilka) [reply packages/good-design/src/hooks/useModalState.ts]: done

File: packages/good-design/src/hooks/useQueryParam.ts (hunk 6 above)
Comment (roma-hladilka) [reply packages/good-design/src/hooks/useQueryParam.ts]: changed

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**